src/PatientChecking/PatientChecking/Feature/Patient/Commands/UpdatePatientInformationCommand.cs

```diff
@@ -24,30 +24,38 @@ public UpdatePatientInformationCommandHandler(IPatientService patientService)
-                MaritalStatus = request.PatientModel.MaritalStatus == (int)PatientMaritalStatus.Married,
-                Gender = request.PatientModel.Gender,
-                EthnicRace = request.PatientModel.Nationality == PatientNationality.Vietnamese.ToString() ? request.PatientModel.EthnicRace : null,
-                HomeTown = request.PatientModel.HomeTown,
-                BirthplaceCity = request.PatientModel.BirthplaceCity,
-                InsuranceNo = request.PatientModel.InsuranceNo,
-                IdcardNo = request.PatientModel.IdcardNo,
-                IssuedDate = !string.IsNullOrEmpty(request.PatientModel.IssuedDate) ? DateTime.Parse(request.PatientModel.IssuedDate) : null,
-                IssuedPlace = request.PatientModel.IssuedPlace
-            };
+                var patientDetails = await _patientService.GetPatientInDetail(request.PatientModel.PatientId);
 
-            var result = await _patientService.UpdatePatientDetail(patientDetails);
+                if (patientDetails != null)
+                {
+                    patientDetails.FirstName = request.PatientModel.FirstName;
+                    patientDetails.MiddleName = request.PatientModel.MiddleName;
+                    patientDetails.LastName = request.PatientModel.LastName;
+                    patientDetails.FullName = string.IsNullOrEmpty(request.PatientModel.MiddleName) ? request.PatientModel.FirstName + " " + request.PatientModel.LastName : request.PatientModel.FirstName + " " + request.PatientModel.MiddleName + " " + request.PatientModel.LastName;
+                    patientDetails.Nationality = request.PatientModel.Nationality;
+                    patientDetails.DoB = DateTime.Parse(request.PatientModel.DoB);
+                    patientDetails.MaritalStatus = request.PatientModel.MaritalStatus == (int)PatientMaritalStatus.Married;
+                    patientDetails.Gender = request.PatientModel.Gender;
+                    patientDetails.EthnicRace = request.PatientModel.Nationality == PatientNationality.Vietnamese.ToString() ? request.PatientModel.EthnicRace : null;
+                    patientDetails.HomeTown = request.PatientModel.HomeTown;
+                    patientDetails.BirthplaceCity = request.PatientModel.BirthplaceCity;
+                    patientDetails.InsuranceNo = request.PatientModel.InsuranceNo;
+                    patientDetails.IdcardNo = request.PatientModel.IdcardNo;
+                    patientDetails.IssuedDate = !string.IsNullOrEmpty(request.PatientModel.IssuedDate) ? DateTime.Parse(request.PatientModel.IssuedDate) : null;
+                    patientDetails.IssuedPlace = request.PatientModel.IssuedPlace;
 
-            return result;
+                    var result = await _patientService.UpdatePatientDetail(patientDetails);
 
+                    return result;
+                }
+                return -1;
+            }
+            catch (FormatException)
+            {
+                return -1;
```

Comment:
I have noticed that if someone inspect the input of this field and change type of this input. It can crash the system. To be safe, I catch this exception. But I'm not sure my assumption is correct. Please correct me @minhphien

Reply:
the application need a global error handling, not this way

Reply:
I have removed this try catch. Please review this @minhphien